onnx/defs/schema.cc

```diff
@@ -107,6 +107,30 @@ OpSchemaRegistry* OpSchemaRegistry::Instance() {
 
 void OpSchema::CheckInputOutputType(struct InferenceContext& ctx) const {
   std::unordered_map<std::string, std::string> type_constraints;
+  if (inputs_.empty() && ctx.getNumInputs() > 0) {
```

Comment:
It would be more general to check that either inputs_ size equals ctx.getNumInputs() or that the last inputs_ is variadic.

Reply:
Created https://github.com/onnx/onnx/issues/5993